core/serialisation/data_serialiser.py

```diff
@@ -78,8 +80,13 @@ def serialise_download_data(
     for sheet in sheets_required:
         query_extender, schema = table_queries[sheet]
         extended_query = query_extender(base_query)
+        # TODO: remove this conditional test on postgres once we strip out sqlite (see FMD-8)
+        if db.engine.name == "postgresql":
+            db.session.execute(text("SET LOCAL work_mem TO '128MB'"))
         table_data = extended_query.all()
         table_serialised = schema(many=True).dump(table_data)
+        if db.engine.name == "postgresql":
```

Comment:
so if this is actually needed, surely we also need to handle it in the case that the query/serialise functionality raises an error, and hence the transaction does not complete atomically?

Reply:
That setting applies on a session level. So if there is an exception, the process will terminate and the db connection will close. Subsequent connections (e.g. by new processes) will use the default setting which is currently 4MB. So no need to revert.

Reply:
hmm, actually probably not since the whole session will end